### PR TITLE
Misc build fixes against latest dev

### PR DESCRIPTION
**Pull request by elrond79**
Fixes two issues I encountered when trying to build on linux:

- cmake was erroring when ${MAYAUSD_DEFINE_BOOST_DEBUG_PYTHON_FLAG} was not defined
- PRIu64 was not defined

### CODE REVIEW

**Comment by HamedSabri-adsk** [comment]
@elrond79 lgtm! 

Was this the error you were getting?

```
CMake Error at lib/CMakeLists.txt:376 (if):
  if given arguments:
    "IS_WINDOWS" "AND"
  Unknown arguments specified
```

@cfmoore007 I believe this is a better fix to the issues that you mentioned earlier on slack. Can you give it a try and see if it solves your issues?

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp [hunk 1]
```diff
@@ -13,6 +13,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+#include <cinttypes>
+
```
Comment (HamedSabri-adsk) [inline plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: Can this be wrapped around some #ifdef? Not sure if this applies to other platforms.

Comment (ppt-adsk) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: Should be fine, it's a C++ 11 header.

Comment (HamedSabri-adsk) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: sounds good!

**Comment by elrond79** [comment]
> Was this the error you were getting?
> 
> ```
> CMake Error at lib/CMakeLists.txt:376 (if):
>   if given arguments:
>     "IS_WINDOWS" "AND"
>   Unknown arguments specified
> ```

Yup, that was it exactly!  Just FYI, you could also fix by adding quotes - "${MAYAUSD_DEFINE_BOOST_DEBUG_PYTHON_FLAG}" - but stylistically I guess prefer stripping the variable deref.

**Comment by HamedSabri-adsk** [comment]
Good to know, I prefer the style you already chose!

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp (hunk 1 above)
Comment (kxl-adsk) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: OK, I understand that fixes done to remove compilation errors on OSX when building in strict mode created this issue - https://github.com/Autodesk/maya-usd/commit/c55c976fcefc07bc44f98c4e808e4089fcb60372#diff-94b4ec7e80f873f5e91609163e535682
Correct?

**Comment by seando-adsk** [comment]
Sorry about the mistake with ${MAYAUSD_DEFINE_BOOST_DEBUG_PYTHON_FLAG} - I don't know why I did that. There is one more in "cmake/defaults/CXXDefaults.cmake" that I think should be changed as well.

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp (hunk 1 above)
Comment (ppt-adsk) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: Yes, and depending on which of inttypes.h or cinttypes is included first, you will get different results --- see
https://stackoverflow.com/questions/14535556/why-doesnt-priu64-work-in-this-code/30851225

**Comment by elrond79** [comment]
Good catch on the other uses of ${MAYAUSD_DEFINE_BOOST_DEBUG_PYTHON_FLAG}, I pushed a updated fix.  And no worries - I've made the same mistake before (as has probably anyone who's written cmake files at some point).

**Comment by elrond79** [comment]
Thanks for the fast turnaround on this!

**Pull request merged**